Audio update

pike983 opened the pull request:
Adding alternative media path searching to the audio file.

pike983 committed:
Added alternative search paths for audio files.

Added alternative search paths for audio files and changed the files to play from the Game and Engine media folders. Added other file types to the CMake so audio files can be included in builds.

pike983 committed:
Update Audio.h

Fixed missed changes in Audio header file.

pike983 committed:
Added option for finding sound via the full path instead of just the sound name for limited directory search.

Added option for finding sound via the full path instead of just the sound name for limited directory search.

pike983 committed:
Merge branch 'develop' into audioUpdate

pike983 committed:
Converted play functions to use filesystem as a backup find method.

Play functions now use a filesystem search in specific directories if an audio file cannot be found right away.

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+		{
+			return theSound;
+		}
+		else
```
this else is redundant because of the return

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+			return nullptr;
+		}
+	}
+	else
```
this else is redundant because of the returns

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+					{
+						return theSound2;
+					}
+					else
```
same here

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+			{
+				if (dEntry.path().filename() == soundNamePath)
+				{
+					ISound* theSound2 = engine->play2D(dEntry.path().string().c_str(), loop, startPaused, true, ESM_AUTO_DETECT, useSoundEffects);
```
This line might be a bit too long, should be shortened according to standards
(Could just show up that way though since it's on github web that looks long. Just double check in IDE)

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+			return nullptr;
+		}
+	}
+	else
```
same with the following elses

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+		}
+		else
+		{
+			string filePath2 = std::filesystem::current_path().parent_path().string() + "\\Game\\src\\Media\\";
```
I suggest having a private member so it's easier to change the dir later on

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+	else
 	{
-		return theSound;
+		string filePath = std::filesystem::current_path().parent_path().string() + "\\Engine\\src\\Media\\";
```
I suggest having a private member so it's easier to change the dir later on

Snow106 commented on Engine/src/Engine/Audio.cpp:
```diff
@@ -19,38 +21,114 @@ AudioPlayerSingleton* AudioPlayerSingleton::getInstance()
+		}
+		else
+		{
+			string filePath2 = std::filesystem::current_path().parent_path().string() + "\\Game\\src\\Media\\";
```
I suggest having a private member so it's easier to change the dir later on

Snow106 commented on Engine/src/Engine/Audio.h:
```diff
@@ -3,6 +3,8 @@
 #define _AUDIO_H_
 
 #include "irrKlang.h"
+#include <string>
```
I don't see either of these includes being used in the header. Consider moving to .cpp

Snow106 reviewed (changes requested):
Please add a description to the PR. It will be easier to understand what was integrated later on.

pike983 committed:
Addressed PR comments.

Addressed the PR comments regarding else statements, line length, member variables and include statements.

Snow106 reviewed (approved):
Nice work!
